### ISSUE TITLE
PO3DSRedirectCustomTabLauncher and POAlternativePaymentMethodCustomTabLauncher are not process death resilient

### ISSUE THREAD

**Issue by Tyomnuy** [processout/processout-android#102]
The current implementation of the [PO3DSRedirectCustomTabLauncher](https://github.com/processout/processout-android/blob/master/sdk/src/main/kotlin/com/processout/sdk/ui/threeds/PO3DSRedirectCustomTabLauncher.kt) and [POAlternativePaymentMethodCustomTabLauncher](https://github.com/processout/processout-android/blob/master/sdk/src/main/kotlin/com/processout/sdk/ui/apm/POAlternativePaymentMethodCustomTabLauncher.kt) doesn't allow resubscribing without calling the `launch()` method again.
If the process is recreated, then it's possible to resubscribe to the activity result by calling `PO3DSRedirectCustomTabLauncher.create()`, but in that case, the `delegate` field will not be initialized, which causes the crash. 

```Kotlin
FATAL EXCEPTION: main
    Process: com.processout.example.staging.debug, PID: 8366
    java.lang.RuntimeException: Unable to create application com.processout.example.ProcessOutExampleApplication: java.lang.ClassCastException
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5876)
        at android.app.ActivityThread.access$1100(ActivityThread.java:199)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1650)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
    Caused by: java.lang.ClassCastException
        at androidx.appcompat.app.ToolbarActionBar$$ExternalSyntheticThrowCCEIfNotNull0.m(R8$$SyntheticClass:0)
        at com.processout.sdk.di.NetworkGraphImpl.<init>(NetworkGraph.kt:58)
        at com.processout.sdk.api.ProcessOut$Companion.configure(ProcessOut.kt:72)
        at com.processout.example.ProcessOutExampleApplication.onCreate(ProcessOutExampleApplication.kt:11)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1154)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5871)
        at android.app.ActivityThread.access$1100(ActivityThread.java:199) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1650) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6669) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 

``` 

Is it possible to move `callback: (ProcessOutResult<String>) -> Unit` from the `launch()` method to the `create()`? To follow the same approach as in the [PONativeAlternativePaymentMethodLauncher](https://github.com/processout/processout-android/blob/master/sdk/src/main/kotlin/com/processout/sdk/ui/nativeapm/PONativeAlternativePaymentMethodLauncher.kt).

**Comment by vitalii-vanziak-cko** [comment]
Hey @Tyomnuy, thanks for the details.
This [PR](https://github.com/processout/processout-android/pull/103) fixes those issues and will be included in the next release (4.7.0).

Callback moved to the `create()` method for `POAlternativePaymentMethodCustomTabLauncher`.

However `PO3DSRedirectCustomTabLauncher` works differently. Callback is not exposed publicly because launcher is not responsible for producing the final result of the 3DS flow. 3DS is a multi-step process and launcher is only one of the steps. Internally we have 3DS service that needs to receive callback from initial launcher invocation once the challenge is done. Updated implementation will handle launcher re-creation using internal cache, so you still need to pass callback to our 3DS service while calling `launch()`.

```
override fun handle(redirect: PO3DSRedirect, callback: (ProcessOutResult<String>) -> Unit) {
        customTabLauncher.launch(
            redirect = redirect,
            returnUrl = "your.application.id://processout/return",
            callback = callback
        )
    }
```
When 3DS service receives callback from launcher it continues the flow by calling backend and cleaning resources before producing the final result. To avoid memory leaks when activity/fragment is re-created the final callback is now decoupled from methods, so it does not hold the reference to initial activity/fragment that invoked it. Instead you will need to subscribe to the SharedFlow in actual coroutines scope for the final result. Here's updated [documentation](https://github.com/processout/processout-android/blob/master/sdk/dokka/3DS.md#3ds2).

**Comment by Tyomnuy** [comment]
We are also using PO3DSRedirectCustomTabLauncher directly for Web Payments (alternative payments like DotPay, Cellulant, etc.), which doesn't involve the 3DS service. But with the cache that you mentioned, it should be covered.
Thank you for the quick and detailed answer!
